cmd/kaspactl/command_parser.go

```diff
@@ -0,0 +1,170 @@
+		valueInt64, err = strconv.ParseInt(valueStr, 10, 32)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+		value = int32(valueInt64)
+	case reflect.Int64:
+		value, err = strconv.ParseInt(valueStr, 10, 64)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+	case reflect.Uint8:
+		var valueUInt64 uint64
+		valueUInt64, err = strconv.ParseUint(valueStr, 10, 8)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+		value = uint8(valueUInt64)
+	case reflect.Uint16:
+		var valueUInt64 uint64
+		valueUInt64, err = strconv.ParseUint(valueStr, 10, 16)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+		value = uint16(valueUInt64)
+	case reflect.Uint32:
+		var valueUInt64 uint64
+		valueUInt64, err = strconv.ParseUint(valueStr, 10, 32)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+		value = uint32(valueUInt64)
+	case reflect.Uint64:
+		value, err = strconv.ParseUint(valueStr, 10, 64)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+	case reflect.Float32:
+		var valueFloat64 float64
+		valueFloat64, err = strconv.ParseFloat(valueStr, 32)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+		value = float32(valueFloat64)
+	case reflect.Float64:
+		value, err = strconv.ParseFloat(valueStr, 64)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+	case reflect.String:
+		value = valueStr
+	case reflect.Struct:
+		pointer := reflect.New(parameterDesc.typeof) // create pointer to this type
+		fieldInterface := pointer.Interface().(proto.Message)
+		err := protojson.Unmarshal([]byte(valueStr), fieldInterface)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+		// Unpointer the value once it's ready
+		fieldInterfaceValue := reflect.ValueOf(fieldInterface)
+		value = fieldInterfaceValue.Elem().Interface()
+	case reflect.Ptr:
+		dummyParameterDesc := &parameterDescription{
+			name:   "valuePointedTo",
+			typeof: parameterDesc.typeof.Elem(),
+		}
+		valuePointedTo, err := stringToValue(dummyParameterDesc, valueStr)
+		if err != nil {
+			return reflect.Value{}, errors.WithStack(err)
+		}
+		pointer := pointerToValue(valuePointedTo)
+
+		value = pointer.Interface()
+
+	// Int and uint are not supported because their size is platform-dependant
+	case reflect.Int:
+	case reflect.Uint:
+	// Other types are not supported simply because they are not used in any command right now
+	// but support can be added if and when needed
+	case reflect.Slice:
+	case reflect.Func:
+	case reflect.Interface:
+	case reflect.Map:
+	case reflect.UnsafePointer:
+	case reflect.Invalid:
+	case reflect.Uintptr:
+	case reflect.Complex64:
+	case reflect.Complex128:
+	case reflect.Array:
+	case reflect.Chan:
+	default:
+		return reflect.Value{},
+			errors.Errorf("Unsupported type '%s' for parameter '%s'", parameterDesc.typeof.Kind(), parameterDesc.name)
+	}
```

Comment:
Do these cases fall through into the default case?(ie error)

Reply:
I was sure it does, but went to check, and found out it does not.
So fixed.